src/main/java/org/fogbowcloud/manager/core/models/orders/Order.java

```diff
@@ -22,18 +25,28 @@
 	private Long fulfilledTime;
 
 	public Order() {
-	
+
 	}
 
-	public Order(OrderState orderState, Token localToken, Token federationToken, String requestingMember,
-			String providingMember, OrderInstance orderInstace, Long fulfilledTime) {
-		this.orderState = orderState;
+	/**
+	 * Creating Order with predefined Id.
+	 */
+	public Order(String id, Token localToken, Token federationToken, String requestingMember, String providingMember) {
+		this.id = id;
+		this.orderState = OrderState.OPEN;
+		this.localToken = localToken;
+		this.federationToken = federationToken;
+		this.requestingMember = requestingMember;
+		this.providingMember = providingMember;
+	}
+
+	public Order(Token localToken, Token federationToken, String requestingMember, String providingMember) {
```

Comment:
[Just an improvement] For this constructor you can reuse the first Order constructor. You can do: 

```java
public Order(Token localToken, Token federationToken, String requestingMember, String providingMember) {
        String id = UUID.randomUUID().toString();
	this(id, localToken, federationToken, requestingMember, providingMember);
}
```